client/blocks/product-selector/index.jsx

```diff
@@ -62,10 +62,19 @@ export class ProductSelector extends Component {
 	getPurchaseByProduct( product ) {
 		const { intervalType, purchases } = this.props;
 		const productSlugs = product.options[ intervalType ];
+		let relatedProductSlugs = {};
+
+		// When viewing monthly tab, we want to consider existing yearly purchases.
+		if ( 'monthly' === intervalType ) {
+			relatedProductSlugs = { ...product.options.yearly };
```

Comment:
How about adding the related products slugs directly to the `productSlugs` here instead of defining a new variable `relatedProductSlugs` and then adding a second `includes()` call in the `return` statement?